File: scripts/refactor_css.js
const fs = require('fs');
const path = require('path');

const frontendDir = path.join(__dirname, 'frontend', 'src');
const pagesDir = path.join(frontendDir, 'styles', 'pages');

if (!fs.existsSync(pagesDir)) {
  fs.mkdirSync(pagesDir, { recursive: true });
}

const premiumCss = fs.readFileSync(path.join(frontendDir, 'premium.css'), 'utf8');

// Extract global variables
const rootMatch = premiumCss.match(/:root\s*\{[\s\S]*?\n\}/);
const globalVars = rootMatch ? rootMatch[0] : '';

const dashboardKeywords = ['.console-stage', '.console-window', '.console-main', '.role-hero', '.stat-card', '.console-panel', '.queue-table', '.queue-row', '.chart-panel', '.activity-panel', '.add-panel', '.detail-meta-panel', '.bar-item', '.clearance-level-badge', '.status-badge-clearance'];
const authKeywords = ['.auth-card', '.auth-form', '.input-group', '.auth-options', '.auth-footer'];
const componentDetailsKeywords = ['.preview-split-stage', '.component-preview-box', '.preview-pane-left', '.preview-theme-dark', '.code-preview-pane', '.code-stage', '.pane-header-mini', '.tab-btn.active', '.icon-button.active-favorite', 'pre[class', 'code[class'];
const viewComponentsKeywords = ['.component-card', '.registry-row', '.semantic-result-card'];

let dashboardCss = '/* Dashboard specific styles */\n\n';
let authCss = '/* Auth (Login/Signup) specific styles */\n\n';
let componentDetailsCss = '/* Component Details specific styles */\n\n';
let viewComponentsCss = '/* View Components specific styles */\n\n';
let globalOverrides = '/* Global Overrides */\n\n';

const rules = [...premiumCss.matchAll(/([^{]+)\{([^}]+)\}/g)];

for (const match of rules) {
  const selector = match[1].trim();
  const body = match[2];
  
  if (selector === ':root' || selector.startsWith('@import')) continue;
  
  const rule = `${selector} {${body}}\n\n`;
  
  if (authKeywords.some(k => selector.includes(k))) {
    authCss += rule;
  } else if (dashboardKeywords.some(k => selector.includes(k))) {
    dashboardCss += rule;
  } else if (componentDetailsKeywords.some(k => selector.includes(k))) {
    componentDetailsCss += rule;
  } else if (viewComponentsKeywords.some(k => selector.includes(k))) {
    viewComponentsCss += rule;
  } else {
    globalOverrides += rule;
  }
}

fs.writeFileSync(path.join(pagesDir, 'Dashboard.css'), dashboardCss);
fs.writeFileSync(path.join(pagesDir, 'Auth.css'), authCss);
fs.writeFileSync(path.join(pagesDir, 'ComponentDetails.css'), componentDetailsCss);
fs.writeFileSync(path.join(pagesDir, 'ViewComponents.css'), viewComponentsCss);
fs.writeFileSync(path.join(pagesDir, 'Admin.css'), '/* Admin styles */\n');
fs.writeFileSync(path.join(pagesDir, 'Favorites.css'), '/* Favorites styles */\n');
fs.writeFileSync(path.join(pagesDir, 'SearchComponents.css'), '/* Search styles */\n');

let appCss = fs.readFileSync(path.join(frontendDir, 'App.css'), 'utf8');
appCss = appCss.replace(/:root(\[data-theme="light"\])?\s*\{[\s\S]*?\n\}/g, '');

const finalGlobalCss = `${globalVars}\n\n${appCss}\n\n${globalOverrides}`;
fs.writeFileSync(path.join(frontendDir, 'styles', 'global.css'), finalGlobalCss);

console.log('CSS files distributed successfully.');

File: scripts/refactor_css.py
import os
import re

frontend_dir = "frontend/src"
pages_dir = os.path.join(frontend_dir, "styles", "pages")
os.makedirs(pages_dir, exist_ok=True)

with open(os.path.join(frontend_dir, "premium.css"), "r") as f:
    premium_css = f.read()

# Base global CSS
global_vars = re.search(r':root\s*\{.*?\n\}', premium_css, re.DOTALL)
global_vars_str = global_vars.group(0) if global_vars else ""

# Categorize classes
dashboard_keywords = [".console-stage", ".console-window", ".console-main", ".role-hero", ".stat-card", ".console-panel", ".queue-table", ".queue-row", ".chart-panel", ".activity-panel", ".add-panel", ".detail-meta-panel", ".bar-item", ".clearance-level-badge", ".status-badge-clearance"]
auth_keywords = [".auth-card", ".auth-form", ".input-group", ".auth-options", ".auth-footer"]
component_details_keywords = [".preview-split-stage", ".component-preview-box", ".preview-pane-left", ".preview-theme-dark", ".code-preview-pane", ".code-stage", ".pane-header-mini", ".tab-btn.active", ".icon-button.active-favorite", "pre[class", "code[class"]
view_components_keywords = [".component-card", ".registry-row", ".semantic-result-card"]
global_keywords = ["body", "h1", "h2", "h3", "h4", "h5", "h6", ".eyebrow", ".brand-logo", "p", "strong", "b", ".premium-props-table", ".secondary-button", ".quick-chip-btn", "button", "input", ".search-input-wrap", ".toast", ".notice", ".console-alert"]

dashboard_css = "/* Dashboard specific styles */\n"
auth_css = "/* Auth (Login/Signup) specific styles */\n"
component_details_css = "/* Component Details specific styles */\n"
view_components_css = "/* View Components specific styles */\n"
global_overrides = "/* Global Overrides */\n"

# A naive parser to split premium.css into rules
rules = re.findall(r'([^{]+)\{([^}]+)\}', premium_css)

for selector, body in rules:
    selector = selector.strip()
    if selector == ":root" or selector.startswith("@import"):
        continue
    
    rule = f"{selector} {{{body}}}\n\n"
    
    if any(k in selector for k in auth_keywords):
        auth_css += rule
    elif any(k in selector for k in dashboard_keywords):
        dashboard_css += rule
    elif any(k in selector for k in component_details_keywords):
        component_details_css += rule
    elif any(k in selector for k in view_components_keywords):
        view_components_css += rule
    else:
        global_overrides += rule

# Write to page files
with open(os.path.join(pages_dir, "Dashboard.css"), "w") as f: f.write(dashboard_css)
with open(os.path.join(pages_dir, "Auth.css"), "w") as f: f.write(auth_css)
with open(os.path.join(pages_dir, "ComponentDetails.css"), "w") as f: f.write(component_details_css)
with open(os.path.join(pages_dir, "ViewComponents.css"), "w") as f: f.write(view_components_css)
with open(os.path.join(pages_dir, "Admin.css"), "w") as f: f.write("/* Admin styles */\n")
with open(os.path.join(pages_dir, "Favorites.css"), "w") as f: f.write("/* Favorites styles */\n")
with open(os.path.join(pages_dir, "SearchComponents.css"), "w") as f: f.write("/* Search styles */\n")

# Process App.css -> global.css
with open(os.path.join(frontend_dir, "App.css"), "r", encoding="utf-8") as f:
    app_css = f.read()

app_css = re.sub(r':root(\[data-theme="light"\])?\s*\{.*?\n\}', '', app_css, flags=re.DOTALL)
final_global_css = f"{global_vars_str}\n{app_css}\n{global_overrides}"

with open(os.path.join(frontend_dir, "styles", "global.css"), "w", encoding="utf-8") as f:
    f.write(final_global_css)

print("CSS files distributed successfully.")
